api/observability/v1/output_types.go

```diff
@@ -0,0 +1,714 @@
+	//
+	// +kubebuilder:validation:Optional
+	Tuning *BaseOutputTuningSpec `json:"tuning,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	AzureMonitor *AzureMonitor `json:"azureMonitor,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	Cloudwatch *Cloudwatch `json:"cloudwatch,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	Elasticsearch *Elasticsearch `json:"elasticsearch,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	GoogleCloudLogging *GoogleCloudLogging `json:"googleCloudLogging,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	HTTP *HTTP `json:"http,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	Kafka *Kafka `json:"kafka,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	LokiStack *LokiStack `json:"lokiStack,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	Splunk *Splunk `json:"splunk,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	Syslog *Syslog `json:"syslog,omitempty"`
+}
+
+// OutputTLSSpec contains options for TLS connections that are agnostic to the output type.
+type OutputTLSSpec struct {
+	// CA can be used to specify a custom list of trusted certificate authorities.
+	//
+	// +kubebuilder:validation:Optional
+	CA *ConfigMapOrSecretKey `json:"ca,omitempty"`
+
+	// Certificate points to the client certificate to use.
+	//
+	// +kubebuilder:validation:Optional
+	Certificate *ConfigMapOrSecretKey `json:"certificate,omitempty"`
+
+	// Key points to the private key of the client certificate.
+	//
+	// +kubebuilder:validation:Optional
+	Key *SecretKey `json:"key,omitempty"`
+
+	// KeyPassphrase points to the passphrase used to unlock the private key.
+	//
+	// +kubebuilder:validation:Optional
+	KeyPassphrase *SecretKey `json:"keyPassphrase,omitempty"`
+
+	// If InsecureSkipVerify is true, then the TLS client will be configured to skip validating server certificates.
+	//
+	// This option is *not* recommended for production configurations.
+	//
+	// +kubebuilder:validation:Optional
+	InsecureSkipVerify bool `json:"insecureSkipVerify,omitempty"`
+
+	// TLSSecurityProfile is the security profile to apply to the output connection.
+	//
+	// +kubebuilder:validation:Optional
+	TLSSecurityProfile *openshiftv1.TLSSecurityProfile `json:"securityProfile,omitempty"`
+}
+
+type URLSpec struct {
+	// URL to send log records to.
+	//
+	// An absolute URL, with a scheme. Valid schemes depend on `type`.
+	// Special schemes `tcp`, `tls`, `udp` and `udps` are used for types that
+	// have no scheme of their own. For example, to send syslog records using secure UDP:
+	//
+	//     { type: syslog, url: udps://syslog.example.com:1234 }
+	//
+	// Basic TLS is enabled if the URL scheme requires it (for example 'https' or 'tls').
+	// The 'username@password' part of `url` is ignored.
+	// Any additional authentication material is in the `secret`.
+	// See the `secret` field for more details.
+	//
+	// +kubebuilder:validation:Required
+	URL string `json:"url"`
```

Comment:
You should add a length limit to this, and use CEL to validate that it is a valid URL